codegen/templates/service_mock.tmpl

```diff
@@ -0,0 +1,160 @@
+{{- $instance := . -}}
+{{- $leafClass := index .DependencyOrder 0 -}}
+{{- $mockType := printf "Mock%sNodes" (title $leafClass) -}}
+
+package {{$instance.PackageInfo.GeneratedPackageAlias}}
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"time"
+
+
+	"github.com/uber/zanzibar/config"
+	"github.com/uber/zanzibar/runtime"
+
+	module "{{$instance.PackageInfo.ModulePackagePath}}"
+)
+
+// MockService interface
+type MockService interface {
+	MakeHTTPRequest(
+		method string,
+		url string,
+		headers map[string]string,
+		body io.Reader,
+	) (*http.Response, error)
+	MakeTChannelRequest(
+		ctx context.Context,
+		thriftService string,
+		method string,
+		headers map[string]string,
+		req, resp zanzibar.RWTStruct,
+	) (bool, map[string]string, error)
+	{{$mockType}}() *module.{{$mockType}}
+	Start()
+	Stop()
+}
+
+type mockService struct {
+	started        bool
+	server         *zanzibar.Gateway
+	{{camel $mockType}}    *module.{{$mockType}}
+	httpClient     *http.Client
+	tChannelClient zanzibar.TChannelCaller
+}
+
+// TestOptions is the info needed to create a test service
+type TestOptions struct {
+	TChannelClientMethods map[string]string
+}
+
+// NewTestService creates a new MockService
```

Comment:
nit: this function has a signature `(MockService, error)` but never returns the error and instead panics. 

If you want it to panic instead of returning the error, it should be called something like `MustCreateTestService`. Otherwise you can create a wrapper like `MustCreateTestService(NewTestService(...))` similar to [`template.Must`](https://golang.org/pkg/text/template/#Must).

Reply:
It's an oversight that no error was returned, I like the `Must` approach, will change it to that.